homeassistant/components/android_ip_webcam/camera.py

```diff
@@ -31,18 +33,45 @@ class IPWebcamCamera(MjpegCamera):
     """Representation of a IP Webcam camera."""
 
     _attr_has_entity_name = True
+    _attr_supported_features = CameraEntityFeature.STREAM
 
     def __init__(self, coordinator: AndroidIPCamDataUpdateCoordinator) -> None:
         """Initialize the camera."""
+        self._cam_username = coordinator.config_entry.data.get(CONF_USERNAME)
+        self._cam_password = coordinator.config_entry.data.get(CONF_PASSWORD, "")
         super().__init__(
             mjpeg_url=coordinator.cam.mjpeg_url,
             still_image_url=coordinator.cam.image_url,
             authentication=HTTP_BASIC_AUTHENTICATION,
-            username=coordinator.config_entry.data.get(CONF_USERNAME),
-            password=coordinator.config_entry.data.get(CONF_PASSWORD, ""),
+            username=self._cam_username,
+            password=self._cam_password,
         )
         self._attr_unique_id = f"{coordinator.config_entry.entry_id}-camera"
         self._attr_device_info = DeviceInfo(
             identifiers={(DOMAIN, coordinator.config_entry.entry_id)},
             name=coordinator.config_entry.data[CONF_HOST],
         )
+        self._coordinator = coordinator
+
+    async def stream_source(self) -> str:
+        """Get the stream source for the Android IP camera."""
+        rtsp_protocol = "rtsp"
+
+        host = self._coordinator.config_entry.data[CONF_HOST]
+        port = self._coordinator.config_entry.data[CONF_PORT]
+
+        video_codec = "h264"  # most compatible & recommended
+        # while "opus" is compatible with more devices,
+        # HA's stream integration requires AAC or MP3,
+        # and IP webcam doesn't provide MP3 audio.
+        # aac is supported on select devices >= android 4.1.
+        # The stream will be quiet on devices that don't support aac,
+        # but it won't fail.
+        audio_codec = "aac"
+
+        if self._cam_username and self._cam_password:
+            return (
+                f"{rtsp_protocol}://{self._cam_username}:{self._cam_password}"
+                f"@{host}:{port}/{video_codec}_{audio_codec}.sdp"
+            )
+        return f"{rtsp_protocol}://{host}:{port}/{video_codec}_{audio_codec}.sdp"
```

Comment:
I think this is simpler and more readable:
```suggestion
        auth = f"self._cam_username:{self._cam_password}@" if self._cam_username else ""
        return f"{rtsp_protocol}://{auth}{host}:{port}/{video_codec}_{audio_codec}.sdp"
```

Reply:
Just noticed I missed the password check, so something like this maybe:
```python
        auth = ""
        if self._cam_username or self._cam_password:
            auth = f"{self._cam_username}:{self._cam_password}@"
        return f"{rtsp_protocol}://{auth}{host}:{port}/{video_codec}_{audio_codec}.sdp"
```

Reply:
@regulad I'm looking at this again and wondering why you didn't add the authentication attributes in the url to the library method when you added it here:

https://github.com/home-assistant-libs/pydroid-ipcam/pull/253